Fix #3583: Merge topic_practice_footer_view into single layout file.

rishidyno opened the pull request:
<!--
  - Thanks for submitting code to Oppia! Please fill out the following as part of
  - your pull request so we can review your code more easily.
  -->

## Explanation

 Fix #3583
<!--
  - Explain what your PR does. If this PR fixes an existing bug, please include
  - "Fixes #bugnum:" in the explanation so that GitHub can auto-close the issue
  - when this PR is merged.
  -->

## Checklist
<!-- Please tick the relevant boxes by putting an "x" in them. -->
- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

| OLD Screenshots         | NEW Screenshots |
| -----------       | -----------      |
|![Screenshot_20210804_152357](https://user-images.githubusercontent.com/78549195/128216905-09edd42a-5b38-4946-a195-e364677f7b6b.png) | ![Screenshot_20210804_152357](https://user-images.githubusercontent.com/78549195/128216927-6016f510-a288-460d-802b-fc1e9e25f228.png) |
| ![Screenshot_20210804_152412](https://user-images.githubusercontent.com/78549195/128216970-e94cf7f2-c081-4751-bc72-9a96cb1cca55.png) | ![Screenshot_20210804_152412](https://user-images.githubusercontent.com/78549195/128216982-df5d79b3-975e-4a67-b09d-7572b87e5433.png) |
| ![Screenshot_20210804_152456](https://user-images.githubusercontent.com/78549195/128217046-324c4968-9b05-4c0f-9c78-546d7bc7b6f5.png) |  ![Screenshot_20210804_152456](https://user-images.githubusercontent.com/78549195/128217084-249a82e2-708f-4a7b-8a0f-1eb565290303.png)|
| ![Screenshot_20210804_152510](https://user-images.githubusercontent.com/78549195/128217103-1304631c-8734-468b-8a51-134a18686ec3.png) | ![Screenshot_20210804_152510](https://user-images.githubusercontent.com/78549195/128217120-c73fdcc6-d030-4e92-a233-8a1b41aa5fd6.png)|

rishidyno committed:
deleted one file of topic practice footer view

rt4914 reviewed (changes requested):
@rishidyno There are 4 files which needs to be merged into 1.

Also as mentioned in issue please add before/after screenshots of this UI in app in mobile portrait/landscape and tablet portrait/landscape mode.

rishidyno commented:
> @rishidyno There are 4 files which needs to be merged into 1.
> 
> Also as mentioned in issue please add before/after screenshots of this UI in app in mobile portrait/landscape and tablet portrait/landscape mode.

@rt4914  I have added Screenshots also I wanted to mention that I tried merging it into one XML file but I ran into some issues.

I added these two in  `mobile portrait/landscape` `dimens.xml` beacause these two are identicle.
<!--  Topic practice footer view-->
  `<dimen name="topic_practice_footer_view_margin_top">36dp</dimen>`
  `<dimen name="topic_practice_footer_view_margin_bottom">108dp</dimen>`

Also i added these two in `tablet portrait/landscape` `dimens.xml` beacause these two are identicle.
<!--  Topic practice footer view-->
  `<dimen name="topic_practice_footer_view_margin_top">24dp</dimen>`
  `<dimen name="topic_practice_footer_view_margin_bottom">0dp</dimen>`

But when I tried to run the app in that activity it crashed in tablet portrait/landscape so I kept the tablet portrait and landscape file.

I could not get a complete understanding of why they crashed. But I think it might be due to other layout elements surrounding them.

I could only delete the landscape file. 

Please guide me on what to do now.

rishidyno committed:
Merge branch 'oppia:develop' into merge-topic_practice_footer_view

rt4914 commented:
> > @rishidyno There are 4 files which needs to be merged into 1.
> > Also as mentioned in issue please add before/after screenshots of this UI in app in mobile portrait/landscape and tablet portrait/landscape mode.
> 
> @rt4914 I have added Screenshots also I wanted to mention that I tried merging it into one XML file but I ran into some issues.
> 
> I added these two in `mobile portrait/landscape` `dimens.xml` beacause these two are identicle.
> 
> `<dimen name="topic_practice_footer_view_margin_top">36dp</dimen>`
> `<dimen name="topic_practice_footer_view_margin_bottom">108dp</dimen>`
> 
> Also i added these two in `tablet portrait/landscape` `dimens.xml` beacause these two are identicle.
> 
> `<dimen name="topic_practice_footer_view_margin_top">24dp</dimen>`
> `<dimen name="topic_practice_footer_view_margin_bottom">0dp</dimen>`
> 
> But when I tried to run the app in that activity it crashed in tablet portrait/landscape so I kept the tablet portrait and landscape file.
> 
> I could not get a complete understanding of why they crashed. But I think it might be due to other layout elements surrounding them.
> 
> I could only delete the landscape file.
> 
> Please guide me on what to do now.

@rishidyno After you have done the changes for by removing the tablet xml file also, rebuild the entire project, uninstall the previous app from tablet device and then install the app. It should work properly after that.

rishidyno committed:
Merge branch 'oppia:develop' into merge-topic_practice_footer_view

rishidyno committed:
Merged `topic_practice_footer_view.xml` into 1 xml file

rishidyno commented:
> > > @rishidyno There are 4 files which needs to be merged into 1.
> > > Also as mentioned in issue please add before/after screenshots of this UI in app in mobile portrait/landscape and tablet portrait/landscape mode.
> > 
> > 
> > @rt4914 I have added Screenshots also I wanted to mention that I tried merging it into one XML file but I ran into some issues.
> > I added these two in `mobile portrait/landscape` `dimens.xml` beacause these two are identicle.
> > `<dimen name="topic_practice_footer_view_margin_top">36dp</dimen>`
> > `<dimen name="topic_practice_footer_view_margin_bottom">108dp</dimen>`
> > Also i added these two in `tablet portrait/landscape` `dimens.xml` beacause these two are identicle.
> > `<dimen name="topic_practice_footer_view_margin_top">24dp</dimen>`
> > `<dimen name="topic_practice_footer_view_margin_bottom">0dp</dimen>`
> > But when I tried to run the app in that activity it crashed in tablet portrait/landscape so I kept the tablet portrait and landscape file.
> > I could not get a complete understanding of why they crashed. But I think it might be due to other layout elements surrounding them.
> > I could only delete the landscape file.
> > Please guide me on what to do now.
> 
> @rishidyno After you have done the changes for by removing the tablet xml file also, rebuild the entire project, uninstall the previous app from tablet device and then install the app. It should work properly after that.

Thanks a lot @rt4914 I followed what you said and I was able to merge it into a single layout file. 
Now it only needs to be reviewed by you.

rishidyno committed:
Delete misc.xml

rishidyno committed:
Merged `topic_practice_footer_view.xml` into 1 xml file

rishidyno committed:
Merge remote-tracking branch 'origin/merge-topic_practice_footer_view' into merge-topic_practice_footer_view

prayutsu commented on .idea/misc.xml:
```diff
@@ -1,9 +0,0 @@
-  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_8" project-jdk-name="1.8" project-jdk-type="JavaSDK">
-    <output url="file://$PROJECT_DIR$/build/classes" />
-  </component>
-  <component name="ProjectType">
```
You only needed to revert the unwanted changes and not delete the file, please restore this file.

rishidyno replied:
I have restored this file

prayutsu commented on .idea/runConfigurations.xml:
```diff
@@ -1,12 +0,0 @@
-        <option value="org.jetbrains.plugins.gradle.execution.test.runner.AllInPackageGradleConfigurationProducer" />
-        <option value="org.jetbrains.plugins.gradle.execution.test.runner.TestClassGradleConfigurationProducer" />
-        <option value="org.jetbrains.plugins.gradle.execution.test.runner.TestMethodGradleConfigurationProducer" />
-      </set>
```
Ditto

rishidyno replied:
and this as well

prayutsu reviewed (changes requested):
Left two comments, PTAL @rishidyno.
Also, note that if you want any reviewers to add their review, you should assign them the PR. Reviewers only add their reviews to the PRs assigned to them.

rishidyno committed:
Merge branch 'oppia:develop' into merge-topic_practice_footer_view

rishidyno committed:
added deleted files misc.xml and runConfigurations.xml

rishidyno committed:
Merge branch 'oppia:develop' into merge-topic_practice_footer_view

rishidyno commented:
Ptal @rt4914 

> Left two comments, PTAL @rishidyno.
> Also, note that if you want any reviewers to add their review, you should assign them the PR. Reviewers only add their reviews to the PRs assigned to them.

Got it 🙂

prayutsu reviewed (approved):
LGTM, Thanks @rishidyno

rishidyno committed:
Merge branch 'oppia:develop' into merge-topic_practice_footer_view

rishidyno committed:
Merge branch 'oppia:develop' into merge-topic_practice_footer_view

rishidyno commented:
> LGTM, Thanks @rishidyno

Thanks @prayutsu

rt4914 commented on app/src/main/res/values-sw600dp-land/dimens.xml:
```diff
@@ -290,4 +290,8 @@
   <dimen name="coming_soon_topic_promoted_list_view_padding_start">0dp</dimen>
   <dimen name="coming_soon_topic_Promoted_list_view_padding_end">0dp</dimen>
 
+  <!--  Topic practice footer view-->
```
Nit spacing missing
`<!--  Topic practice footer view -->`

rishidyno replied:
@rt4914  I have added spaces and capitalized this.

rt4914 commented on app/src/main/res/values-sw600dp-port/dimens.xml:
```diff
@@ -293,4 +293,8 @@
   <dimen name="coming_soon_topic_promoted_list_view_padding_start">0dp</dimen>
   <dimen name="coming_soon_topic_Promoted_list_view_padding_end">0dp</dimen>
 
+  <!--  Topic practice footer view-->
```
Nit spacing missing
`<!--  Topic practice footer view -->`

rishidyno replied:
same

rt4914 commented on app/src/main/res/values/dimens.xml:
```diff
@@ -450,4 +450,8 @@
   <dimen name="coming_soon_topic_promoted_list_view_padding_start">28dp</dimen>
   <dimen name="coming_soon_topic_Promoted_list_view_padding_end">28dp</dimen>
 
+  <!--  Topic practice footer view-->
```
Nit spacing missing
`<!--  Topic practice footer view -->`

rishidyno replied:
done

rt4914 reviewed (changes requested):
@rishidyno Code is good. Just nit changes suggested.

rishidyno committed:
Merge branch 'oppia:develop' into merge-topic_practice_footer_view

rishidyno committed:
added space and capitalized the letters

rt4914 reviewed (approved):
LGTM, thanks.
